prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -87,17 +87,36 @@ fun Observation.addMappedConditions(metadata: Metadata): List<ActionLogDetail> {
     val codeSourcesMap = this.getCodeSourcesMap().filterValues { it.isNotEmpty() }
     var mappedSomething = false
     if (codeSourcesMap.values.flatten().isEmpty()) return listOf(UnmappableConditionMessage()) // no codes found
-
+    val mappingTable = metadata.findLookupTable("observation-mapping").also {
```

Comment:
The optimization here is to use the table saw filtering mechanism as well as fetching all the rows that are applicable for the observation.  There could be a further optimization where this is actually done for the whole bundle, but I'm leaving it like this for now, since It seems clearer